xds/internal/xdsclient/xdsresource/unmarshal_rds_test.go

```diff
@@ -575,6 +615,72 @@ func (s) TestRDSGenerateRDSUpdateFromRouteConfiguration(t *testing.T) {
 			wantUpdate: goodUpdateIfRetryDisabled(),
 			wantError:  env.RetrySupport,
 		},
+		{
+			name: "cluster-specifier-declared-which-not-registered",
+			rc: goodRouteConfigWithClusterSpecifierPlugins([]*v3routepb.ClusterSpecifierPlugin{
+				{
+					Extension: &v3corepb.TypedExtensionConfig{
+						Name:        "cspA",
+						TypedConfig: configOfClusterSpecifierDoesntExist,
+					},
+				},
+			}, []string{"cspA"}),
+			wantError: true,
+		},
+		{
+			name: "error-in-cluster-specifier-plugin-conversion-method",
+			rc: goodRouteConfigWithClusterSpecifierPlugins([]*v3routepb.ClusterSpecifierPlugin{
+				{
+					Extension: &v3corepb.TypedExtensionConfig{
+						Name:        "cspA",
+						TypedConfig: errorClusterSpecifierConfig,
+					},
+				},
+			}, []string{"cspA"}),
+			wantError: true,
+		},
+		{
+			name: "route-action-that-references-undeclared-cluster-specifier-plugin",
+			rc: goodRouteConfigWithClusterSpecifierPlugins([]*v3routepb.ClusterSpecifierPlugin{
+				{
+					Extension: &v3corepb.TypedExtensionConfig{
+						Name:        "cspA",
+						TypedConfig: mockClusterSpecifierConfig,
+					},
+				},
+			}, []string{"cspA", "cspB"}),
+			wantError: true,
+		},
+		{
+			name: "emitted-cluster-specifier-plugins",
+			rc: goodRouteConfigWithClusterSpecifierPlugins([]*v3routepb.ClusterSpecifierPlugin{
+				{
+					Extension: &v3corepb.TypedExtensionConfig{
+						Name:        "cspA",
+						TypedConfig: mockClusterSpecifierConfig,
+					},
+				},
```

Comment:
Declare this as a variable since you use it many times?  Or a function that takes `name` and `typedConfig` as parameters.  Then you can write in 1 line what otherwise takes 6.

Reply:
Declared this as a function that takes name and typedConfig.